modules/cpmso-creating-cr.adoc

```diff
@@ -6,4 +6,88 @@
\ No newline at end of file
+To use the Control Plane Machine Set Operator, you must ensure that a `ControlPlaneMachineSet` custom resource (CR) with the correct settings for your cluster exists. On a cluster without a generated CR, you must create the CR manually and activate the Operator.
+
+[NOTE]
+====
+For more information about the structure and parameters of the CR, see "Control Plane Machine Set Operator configuration".
+====
+
+.Procedure
+
+. Create a YAML file using the following template:
++
+--
+.Control plane machine set CR YAML file template
+[source,yaml]
+----
+apiVersion: machine.openshift.io/v1
+kind: ControlPlaneMachineSet
+metadata:
+  name: cluster
+  namespace: openshift-machine-api
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: <cluster_id> <1>
+      machine.openshift.io/cluster-api-machine-role: master
+      machine.openshift.io/cluster-api-machine-type: master
+  state: Active <2>
+  strategy:
+    type: RollingUpdate <3>
+  template:
+    machineType: machines_v1beta1_machine_openshift_io
+    machines_v1beta1_machine_openshift_io:
+      failureDomains:
+        platform: <platform> <4>
+        <platform_failure_domains> <5>
+      metadata:
+        labels:
+          machine.openshift.io/cluster-api-cluster: <cluster_id> <6>
+          machine.openshift.io/cluster-api-machine-role: master
+          machine.openshift.io/cluster-api-machine-type: master
+      spec:
+        providerSpec:
+          value:
+            <platform_provider_spec> <7>
+----
+<1> Specify the infrastructure ID that is based on the cluster ID that you set when you provisioned the cluster. You must specify this value when you create a `ControlPlaneMachineSet` CR. If you have the OpenShift CLI (`oc`) installed, you can obtain the infrastructure ID by running the following command:
++
+[source,terminal]
+----
+$ oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster
+----
+<2> Specify the state of the Operator. When the state is `Inactive`, the Operator is not operational. You can activate the Operator by setting the value to `Active`. 
++
+[IMPORTANT]
+====
+Before you activate the Operator, you must ensure that the CR configuration is correct for your cluster requirements.
+====
+<3> Specify the update strategy for the cluster. The allowed values are `OnDelete` and `RollingUpdate`. The default value is `RollingUpdate`. 
+//For more information about update strategies, see "Updating the control plane configuration".
+<4> Specify your cloud provider platform name. The allowed values are `AWS`, `Azure`, and `VSphere`. 
+<5> Add the `<platform_failure_domains>` configuration for the cluster. The format and values of this section are provider-specific. For more information, see the sample failure domain configuration for your cloud provider.
++
+[NOTE]
+====
+VMware vSphere does not support failure domains.
+====
```

Comment:
I haven't personally tested this, what do we want the user to do in this case @JoelSpeed? Omit the field?

Reply:
I think I got all feedback addressed. Will wait on this answer before merge. Thanks, everyone!

Reply:
QE tested this and AFAIK they just omitted the entire `failureDomains` section, that said, i think you could include the platform type and it wouldn't error. Perhaps @sunzhaohua2 can help check what QE tested

Reply:
@JoelSpeed , here is how it was the failure domains were not populated they were left like 
`failureDomains:`
for reference - 
cpms [yaml](https://gist.github.com/miyadav/f5046cbba46c3ff1a9831a25c7bc18d7) for vsphere 
related [case](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-53099) 

so LGTM too from documentation POV.

Reply:
ok, since they can omit I will state that as it seems least error-prone